source/includes/read/change-streams.php

```diff
@@ -0,0 +1,47 @@
+<?php
+require 'vendor/autoload.php'; 
+
+$uri = getenv('MONGODB_URI') ?: throw new RuntimeException('Set the MONGODB_URI variable to your Atlas URI that connects to the sample dataset');
+$client = new MongoDB\Client($uri);
+
+// start-db-coll
+$collection = $client->sample_restaurants->restaurants;
+// end-db-coll
+
+// Monitors and prints changes to the "restaurants" collection
+// start-open-change-stream
+$changeStream = $collection->watch();
+
+foreach ($changeStream as $event) {
```

Comment:
`foreach` loops are not suitable for iterating tailable cursors (such as those returned for change streams), as the loop will terminate as soon as there are no documents to consume. In other words, `foreach` prevents us from actually polling the cursor and waiting for additional data.

This is why the PHPLIB docs had a [Tailable Cursor Iteration](https://www.mongodb.com/docs/php-library/current/tutorial/tailable-cursor/) tutorial. The existing examples in the [`watch()` API reference](https://www.mongodb.com/docs/php-library/current/reference/method/MongoDBCollection-watch/#examples) demonstrate an equivalent approach, with added code to break out of the loop should we receive an "invalidate" event.

I'm not sure if this script is ever intended to be executed in its entirety, or simply serve as an assortment of snippets to include in the rendered docs. But if you're using the proper approach to iterate the change stream, note that the first loop is unlikely to terminate on its own and that would prevent subsequent statements in the file from executing.

Reply:
I changed these loops to for loops and edited based on the examples you linked - let me know what you think